livy/src/main/java/org/apache/zeppelin/livy/BaseLivyInterprereter.java

```diff
@@ -229,7 +229,31 @@ private InterpreterResult getResultFromStatementInfo(StatementInfo stmtInfo,
     } else {
       //TODO(zjffdu) support other types of data (like json, image and etc)
       String result = stmtInfo.output.data.plain_text;
-      if (result != null) {
+      Object magicResult = null;
+      
+      // check magic first
+      if ((magicResult = stmtInfo.output.data.application_livy_table_json) != null) {
```

Comment:
Thanks for trying, but if this is too complicated and not worth to do, then just wrap it in a function for easy unit test.